### ISSUE TITLE
Svelte5 support

### ISSUE THREAD

**Issue by stalkerg** [cibernox/svelte-intl-precompile#67]
Seems like we should replace storage by value.

**Comment by cibernox** [comment]
I need to take a look at the needed changes to support svelte 5

**Comment by stalkerg** [comment]
yeah, the main change it's storage replaced by rune. Shouldn't be too much changes.

**Comment by stalkerg** [comment]
Will be good to do this #66 during this support as well.

**Comment by stalkerg** [comment]
Svelte5 is released. Any updates? Should I help, or should we start searching for a new library?

**Comment by cibernox** [comment]
Now that svelte5 is out I'll find some time to  work on it. I haven't used svelte 5 new idioms a lot yet, so I need to get familiar with them first.

If anyone has used them for a while any input is welcomed.

**Comment by stalkerg** [comment]
As I said, Svelte5 provided a universal reactive solution - $state(), which replaces Svelte4 Stores and component-level variables. I think we just should replace  Store by $state, that's it.  
Basically, `import { t } from 'svelte-intl-precompile';` `t` here should state. 
As I understand, everything else can be kept as is. Of course, internally, you will work with the new $state a little bit differently, but it looks like nothing complicated. 

But we definitely should also solve #66

**Comment by cibernox** [comment]
ok, I started working on this a bit. To make this better the changes are a bit more profound than that. Ideally.

A lot of the code is now unnecesarily complicated because in JS/TS files we had no reactivity, but now with $state being usable outside svelte components the codebase can be greatly simplified.

**Comment by zvikasdongre** [comment]
Hi, Is there any update on the svelte 5 support?

**Comment by macmillen** [comment]
any update?

**Comment by stalkerg** [comment]
I made a fork. API is very similar, but instead of storage, it uses states, and instead of a global object for SSR to avoid a race condition, I am using page contexts. 

https://github.com/stalkerg/svelte-icu
